report/topology.go

```diff
@@ -186,14 +201,19 @@ func (n Nodes) Merge(other Nodes) Nodes {
 		return n
 	}
 	cp := n.Copy()
+	cp.UnsafeMerge(other)
+	return cp
+}
+
+// UnsafeMerge merges the other object into this one, modifying the original.
+func (n *Nodes) UnsafeMerge(other Nodes) {
 	for k, v := range other {
-		if n, ok := cp[k]; ok { // don't overwrite
-			cp[k] = v.Merge(n)
+		if existing, ok := (*n)[k]; ok { // don't overwrite
+			(*n)[k] = v.Merge(existing)
 		} else {
-			cp[k] = v
+			(*n)[k] = v
```

Comment:
I reckon the pointering here hinders rather than helps comprehension - I am reading this and wondering "why is this necessary?".

Reply:
I made the receiver a pointer to stress that it modifies `n`.
Since `n` is really a `map` this isn't necessary.  I'm uncertain which is better.